python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -0,0 +1,750 @@
+			contextCopy = Gaffer.Context( self.getContext() )
+			self.__pathListing.setPath( _GafferSceneUI._RenderPassEditor.RenderPassPath( self.__settingsNode["in"], contextCopy, "/", filter = self.__filter ) )
+		else :
+			self.__pathListing.setPath( Gaffer.DictPath( {}, "/" ) )
+
+	def __buttonDoubleClick( self, pathListing, event ) :
+
+		# A small corner area below the vertical scroll bar may pass through
+		# to us, causing odd selection behavior. Check that we're within the
+		# scroll area.
+		if pathListing.pathAt( event.line.p0 ) is None :
+			return False
+
+		if event.button == event.Buttons.Left :
+			column = pathListing.columnAt( event.line.p0 )
+			if column == self.__renderPassActiveColumn :
+				self.__setActiveRenderPass( pathListing )
+			else :
+				self.__editSelectedCells( pathListing )
+
+			return True
+
+		return False
+
+	def __keyPress( self, pathListing, event ) :
+
+		if event.modifiers == event.Modifiers.None_ :
+
+			if event.key == "Return" or event.key == "Enter" :
+				selection = pathListing.getSelection()
+				if len( selection[1].paths() ) :
+					self.__setActiveRenderPass( pathListing )
+				else :
+					self.__editSelectedCells( pathListing )
+				return True
+
+			if event.key == "D" and len( self.__disablableInspectionTweaks( pathListing ) ) > 0 :
+				self.__disableEdits( pathListing )
+				return True
+
+	def __selectedRenderPasses( self, columns = [ 0 ] ) :
+
+		# There may be multiple columns with a selection, but we only operate on the specified column indices.
+		selection = self.__pathListing.getSelection()
+		renderPassPath = self.__pathListing.getPath().copy()
+		result = set()
+		for c in columns :
+			for path in selection[c].paths() :
+				renderPassPath.setFromString( path )
+				name = renderPassPath.property( "renderPassPath:name" )
+				if name is not None :
+					result.add( name )
+
+		return list( result )
+
+	def __setActiveRenderPass( self, pathListing ) :
+
+		selectedPassNames = self.__selectedRenderPasses( columns = [ 1 ] )
+
+		if len( selectedPassNames ) != 1 :
+			return
+
+		script = self.scriptNode()
+		if Gaffer.MetadataAlgo.readOnly( script ) :
+			with GafferUI.PopupWindow() as self.__popup :
+				with GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing = 4 ) :
+					GafferUI.Image( "warningSmall.png" )
+					GafferUI.Label( "<h4>The script is read-only.</h4>" )
+
+			self.__popup.popup()
+			return
+
+		if "renderPass" not in script["variables"] :
+			renderPassPlug = Gaffer.NameValuePlug( "renderPass", "", "renderPass", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+			script["variables"].addChild( renderPassPlug )
+			Gaffer.MetadataAlgo.setReadOnly( renderPassPlug["name"], True )
+		else :
+			renderPassPlug = script["variables"]["renderPass"]
+
+		renderPassPlug["value"].setValue( selectedPassNames[0] )
+
+	def __editSelectedCells( self, pathListing, quickBoolean = True ) :
```

Comment:
How much - if at all - does this interaction handling code differ from LightEditor? I'm hoping it's identical and we can consolidate it into one in future. If it is, could you add a todo to that effect? And if it's not, it'd be good to understand why...

Reply:
The main difference is the inspectors here require `renderPass` to be set in the context before inspection rather than `scene:path`. I've added a todo for eventual consolidation in 866dd8c

Reply:
OK, cool. I think what this means is that we need to define a way to get the inspection context out of RenderPassPath and ScenePath in a uniform way - does that sound right?